test/integration/interrupt_resume_test.rb

```diff
@@ -104,4 +104,172 @@ def test_interrupt_resume_when_table_has_completed
+
+    # This row would have been copied as we terminate ghostferry after 1 batch
+    # is copied.
+    result = source_db.query("SELECT MIN(id) FROM #{DEFAULT_FULL_TABLE_NAME}")
+    chosen_id = result.first["MIN(id)"] + 1
+
+    i = 0
+    ghostferry.on_status(Ghostferry::Status::AFTER_ROW_COPY) do
+      if i == 1
+        # We need to delete it the second batch because trying to delete the
+        # minimum id row while in the first batch will result in a lock as the
+        # DataIterator holds a FOR UPDATE lock for the minimum id row.
+        source_db.query("DELETE FROM #{DEFAULT_FULL_TABLE_NAME} WHERE id = #{chosen_id}")
+      end
+      sleep if i > 1 # block the DataIterator so it doesn't race with the term_and_wait_for_exit below.
+      i += 1
+    end
+
+    ghostferry.on_status(Ghostferry::Status::AFTER_BINLOG_APPLY) do
+      ghostferry.term_and_wait_for_exit
+    end
+
+    dumped_state = ghostferry.run_expecting_interrupt
+    assert_basic_fields_exist_in_dumped_state(dumped_state)
+    refute_nil dumped_state["BinlogVerifyStore"]
+    refute_nil dumped_state["BinlogVerifyStore"]["gftest"]
+    refute_nil dumped_state["BinlogVerifyStore"]["gftest"]["test_table_1"]
+
+    # FLAKY: See explanation in test_interrupt_inline_verifier_will_emit_binlog_verify_store
+    assert_equal 1, dumped_state["BinlogVerifyStore"]["gftest"]["test_table_1"]["#{chosen_id}"]
+
+    # Corrupt the row before resuming
+    target_db.query("REPLACE INTO #{DEFAULT_FULL_TABLE_NAME} (id, data) VALUES (#{chosen_id}, 'corrupted')")
+
+    verification_ran = false
+    incorrect_tables = nil
+    ghostferry = new_ghostferry(MINIMAL_GHOSTFERRY, config: { verifier_type: "Inline" })
+    ghostferry.on_status(Ghostferry::Status::VERIFIED) do |*tables|
+      verification_ran = true
+      incorrect_tables = tables
+    end
+
+    ghostferry.run(dumped_state)
+    assert verification_ran
+    assert_equal 1, incorrect_tables.length
+    assert_equal "gftest.test_table_1", incorrect_tables.first
+
+    error_line = ghostferry.error_lines.last
+    assert_equal "cutover verification failed for: gftest.test_table_1 [pks: #{chosen_id} ] ", error_line["msg"]
+  end
+
+  def test_interrupt_resume_inline_verifier_will_verify_additional_rows_changed_on_source_during_interrupt
+    ghostferry = new_ghostferry(MINIMAL_GHOSTFERRY, config: { verifier_type: "Inline" })
+
+    ghostferry.on_status(Ghostferry::Status::AFTER_ROW_COPY) do
+      ghostferry.term_and_wait_for_exit
+    end
+
+    dumped_state = ghostferry.run_expecting_interrupt
+    assert_basic_fields_exist_in_dumped_state(dumped_state)
+    refute_nil dumped_state["BinlogVerifyStore"]
+
+    result = source_db.query("SELECT MIN(id) FROM #{DEFAULT_FULL_TABLE_NAME}")
+    chosen_id = result.first["MIN(id)"]
+
+    source_db.query("UPDATE #{DEFAULT_FULL_TABLE_NAME} SET data = 'data2' WHERE id = #{chosen_id}")
+    target_db.query("UPDATE #{DEFAULT_FULL_TABLE_NAME} SET data = 'corrupted' WHERE id = #{chosen_id}")
```

Comment:
> hmmm.. thinking more about... is it because we want the target to not accept the binlog copier's attempt to SET data = 'data2' thinking

This is correct.

This test is to make sure that a binlog event that occured while Ghostferry has been interrupted is correctly picked up. Now, given that Ghostferry should correctly copy the row over upon resume, we'd have no way to observe that the InlineVerifier has also verified this row. Thus, if we set the data to corrupted here, the action of the InlineVerifier becomes observable.

Comment has been added.